packages/rendermime/src/widgets.ts

```diff
@@ -313,6 +313,20 @@ export class RenderedMarkdown extends RenderedHTMLCommon {
     });
   }
 
+  async renderModel(model: IRenderMime.IMimeModel): Promise<void> {
+    // Toggle the trusted class on the widget.
+    this.toggleClass('jp-mod-trusted', model.trusted);
+
+    // Render the actual content.
+    await this.render(model);
+
+    // Handle the fragment identifier if given.
+    const { fragment } = model.metadata;
+    if (fragment) {
+      this.setFragment(fragment as string);
+    }
```

Comment:
Thank you, this solution looks good to me! On implementation level, what if we added a second argument called `clear=true` to `RenderedCommon.renderModel` and then in here call `super.renderModel(model, false)` to avoid duplicating code (and incidental drift in the future)?

Reply:
Thanks for your advice. `RenderedCommon.renderModel` is an implementation of `IRenderMime.IRenderer`, which is not responsible for providing such options. Instead, I add another method with the `clearExising` option in `RenderedCommon`.

Reply:
True, but we can add optional arguments at implementation level without modifying the interface. I am fine with the proposed solution, but let's make `renderCommonModel` protected.

Reply:
That makes sense. I thought an optional arguments would break the lint rule. I just tried it and it should look cleaner now